test/plugin/test_exception_detector.rb

```diff
@@ -350,4 +350,28 @@ def test_json_messages
       assert_equal(expected_json(s.actual_field, s.output), out, s.desc)
     end
   end
+
+  def test_max_lines_limit
+    max_lines = JAVA_EXC_PART1.lines.length
+    out = []
+    buffer = Fluent::TraceAccumulator.new(nil,
+                                          [:all],
+                                          max_lines: max_lines) do |_, m|
+      out << m
+    end
+    feed_lines(buffer, JAVA_EXC)
+    assert_equal([JAVA_EXC_PART1] + JAVA_EXC_PART2.lines, out)
+  end
+
+  def test_max_bytes_limit
+    max_bytes = JAVA_EXC_PART1.length + JAVA_EXC_PART2.lines[0].length - 1
```

Comment:
You might want to add a comment, something along the lines of "_Make sure the trace is flushed after the first part._" (also helps in `test_max_lines_limit`, but here it's actually needed, IMO).
I would also test with `JAVA_EXC_PART1.length + 1`.